lib/compiler/src/engine/artifact.rs

```diff
@@ -26,7 +41,15 @@ use wasmer_vm::{FunctionBodyPtr, MemoryStyle, TableStyle, VMSharedSignatureIndex
 use wasmer_vm::{InstanceAllocator, InstanceHandle, StoreObjects, TrapHandlerFn, VMExtern};
 
 /// A compiled wasm module, ready to be instantiated.
-pub struct Artifact {
+pub enum Artifact {
+    /// FIXME
+    Universal(UniversalArtifact),
+    /// FIXME
+    Static(StaticArtifact),
```

Comment:
I don't think this is the optimal way of doing this. Basically both of them shared the finished functions, signatures and so on.

The only different thing is `frame_info_registration` and `_symbol_registry`. But that's only needed when using loading the artifact. In case of `frame_info_registration` is already optional, and perhaps we can make _symbol_registry is only needed at loading time (not really needed to store it)

Reply:
No it isn't the optimal way, but it was the fastest way to get it working, next step is making it prettier as you say